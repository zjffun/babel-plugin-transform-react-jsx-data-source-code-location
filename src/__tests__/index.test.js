const babel = require("@babel/core");
const plugin = require("../../");

const babelConfig = {
  filename: "/fake/path/mock.js",
  presets: ["@babel/preset-react"],
  plugins: [plugin],
};

it("basic-sample should work", () => {
  const example = `
var x = <div></div>;
var x = <View></View>;
`;
  const { code } = babel.transform(example, babelConfig);
  expect(code).toMatchSnapshot();
});

it("no-jsx will do nothing", () => {
  const example = `
var x = 42;
`;
  const { code } = babel.transform(example, babelConfig);
  expect(code).toMatchSnapshot();
});

it("with data-__srouce-code-location should work", () => {
  const example = `
var x = <img data-__srouce-code-location="" />;
var x = <img data-__srouce-code-location={""} />;
var x = <img data-__srouce-code-location={0} />;
var x = <img data-__srouce-code-location={1} />;
var x = <img data-__srouce-code-location={null} />;
var x = <img data-__srouce-code-location={undefined} />;
var x = <img data-__srouce-code-location={{test: test}} />;
var x = <img data-__srouce-code-location={obj} />;
var x = <img data-__srouce-code-location={obj} data-__srouce-code-location="" />;
`;
  const { code } = babel.transform(example, babelConfig);
  expect(code).toMatchSnapshot();
});

it("with JSXSpreadAttribute should work", () => {
  const example = `
var x = <div {...test}></div>;
var x = <View {...test}></View>;
`;
  const { code } = babel.transform(example, babelConfig);
  expect(code).toMatchSnapshot();
});

it("React.Fragment should not add data-__srouce-code-location", () => {
  const example = `
var x = <></>;
var x = <React.Fragment></React.Fragment>;
`;
  const { code } = babel.transform(example, babelConfig);
  expect(code).toMatchSnapshot();
});

