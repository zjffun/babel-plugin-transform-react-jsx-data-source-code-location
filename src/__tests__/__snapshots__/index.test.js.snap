// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React.Fragment should not add data-__srouce-code-location 1`] = `
"var x = /*#__PURE__*/React.createElement(React.Fragment, null);
var x = /*#__PURE__*/React.createElement(React.Fragment, null);"
`;

exports[`basic-sample should work 1`] = `
"var x = /*#__PURE__*/React.createElement("div", {
  "data-__source-code-location": "/fake/path/mock.js:2:9"
});
var x = /*#__PURE__*/React.createElement(View, {
  "data-__source-code-location": "/fake/path/mock.js:3:9"
});"
`;

exports[`no-jsx will do nothing 1`] = `"var x = 42;"`;

exports[`with JSXSpreadAttribute should work 1`] = `
"function _extends() { return _extends = Object.assign ? Object.assign.bind() : function (n) { for (var e = 1; e < arguments.length; e++) { var t = arguments[e]; for (var r in t) ({}).hasOwnProperty.call(t, r) && (n[r] = t[r]); } return n; }, _extends.apply(null, arguments); }
var x = /*#__PURE__*/React.createElement("div", _extends({}, test, {
  "data-__source-code-location": "/fake/path/mock.js:2:9"
}));
var x = /*#__PURE__*/React.createElement(View, _extends({}, test, {
  "data-__source-code-location": "/fake/path/mock.js:3:9"
}));"
`;

exports[`with data-__srouce-code-location should work 1`] = `
"var x = /*#__PURE__*/React.createElement("img", {
  "data-__srouce-code-location": "",
  "data-__source-code-location": "/fake/path/mock.js:2:9"
});
var x = /*#__PURE__*/React.createElement("img", {
  "data-__srouce-code-location": "",
  "data-__source-code-location": "/fake/path/mock.js:3:9"
});
var x = /*#__PURE__*/React.createElement("img", {
  "data-__srouce-code-location": 0,
  "data-__source-code-location": "/fake/path/mock.js:4:9"
});
var x = /*#__PURE__*/React.createElement("img", {
  "data-__srouce-code-location": 1,
  "data-__source-code-location": "/fake/path/mock.js:5:9"
});
var x = /*#__PURE__*/React.createElement("img", {
  "data-__srouce-code-location": null,
  "data-__source-code-location": "/fake/path/mock.js:6:9"
});
var x = /*#__PURE__*/React.createElement("img", {
  "data-__srouce-code-location": undefined,
  "data-__source-code-location": "/fake/path/mock.js:7:9"
});
var x = /*#__PURE__*/React.createElement("img", {
  "data-__srouce-code-location": {
    test: test
  },
  "data-__source-code-location": "/fake/path/mock.js:8:9"
});
var x = /*#__PURE__*/React.createElement("img", {
  "data-__srouce-code-location": obj,
  "data-__source-code-location": "/fake/path/mock.js:9:9"
});
var x = /*#__PURE__*/React.createElement("img", {
  "data-__srouce-code-location": obj,
  "data-__srouce-code-location": "",
  "data-__source-code-location": "/fake/path/mock.js:10:9"
});"
`;
